src/ansible_navigator/utils/serialize.py

```diff
@@ -63,7 +122,11 @@ class YamlStyle(NamedTuple):
     allow_unicode: bool = True
 
 
-def human_dump(obj: Any, filename: str = None, file_mode: str = "w") -> Optional[str]:
+def human_dump(
+    obj: Any,
+    filename: Optional[Union[Path, str]] = None,
```

Comment:
I'm almost sure that `Optional` is implicit when the default is `None`. So this should be enough:
```suggestion
    filename: Union[Path, str] = None,
```

Reply:
You are 100% right, but it is discouraged:

"This is no longer the recommended behavior. Type checkers should move towards requiring the optional type to be made explicit."

https://www.python.org/dev/peps/pep-0484/#union-types

Reply:
We'll enforce it at some point: https://github.com/ansible/ansible-navigator/issues/1048

Reply:
:nauseated_face: implicit Optional sounds horrible. Please keep the `Optional` here.

(Also I tested locally and mypy fails without explicit Optional, as it should.)

I'm fine with type inference. I'm less fine with behind the scenes magic.

Reply:
It is enforced now: https://github.com/ansible/ansible-navigator/pull/1050